$MirOS: ports/print/cups/patches/patch-backend_Makefile,v 1.2 2006/11/08 18:25:44 bsiegert Exp $
--- backend/Makefile.orig	Wed Aug  8 21:27:28 2007
+++ backend/Makefile	Tue Oct 21 22:56:22 2008
@@ -89,7 +89,7 @@ uninstall:
 
 betest:	betest.o ../cups/$(LIBCUPS)
 	echo Linking $@...
-	$(CC) $(LDFLAGS) -o betest betest.o $(LIBS)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o betest betest.o $(LIBS)
 
 
 #
@@ -98,7 +98,7 @@ betest:	betest.o ../cups/$(LIBCUPS)
 
 test1284:	test1284.o ../cups/libcups.a
 	echo Linking $@...
-	$(CC) $(LDFLAGS) -o test1284 test1284.o ../cups/libcups.a \
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o test1284 test1284.o ../cups/libcups.a \
 		$(SSLLIBS) $(COMMONLIBS) $(LIBZ)
 
 
@@ -119,7 +119,7 @@ libbackend.a:	$(LIBOBJS)
 
 ieee1394:	ieee1394.o ieee1394-linux.o
 	echo Linking $@...
-	$(CC) $(LDFLAGS) -o ieee1394 ieee1394.o ieee1394-linux.o -lraw1394 $(LIBS)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o ieee1394 ieee1394.o ieee1394-linux.o -lraw1394 $(LIBS)
 
 ieee1394.o:		ieee1394.h
 ieee1394-linux.o:	ieee1394.h
@@ -131,7 +131,7 @@ ieee1394-linux.o:	ieee1394.h
 
 ipp:	ipp.o ../cups/$(LIBCUPS)
 	echo Linking $@...
-	$(CC) $(LDFLAGS) -o ipp ipp.o $(LIBS)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o ipp ipp.o $(LIBS)
 	$(RM) http
 	$(LN) ipp http
 
@@ -142,7 +142,7 @@ ipp:	ipp.o ../cups/$(LIBCUPS)
 
 lpd:	lpd.o ../cups/$(LIBCUPS)
 	echo Linking $@...
-	$(CC) $(LDFLAGS) -o lpd lpd.o $(LIBS)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o lpd lpd.o $(LIBS)
 
 
 #
@@ -151,7 +151,7 @@ lpd:	lpd.o ../cups/$(LIBCUPS)
 
 pap:	pap.o ../cups/$(LIBCUPS)
 	echo Linking $@...
-	$(CC) $(LDFLAGS) -o pap pap.o $(BACKLIBS) $(LIBS) -framework AppleTalk
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o pap pap.o $(BACKLIBS) $(LIBS) -framework AppleTalk
 
 
 #
@@ -160,7 +160,7 @@ pap:	pap.o ../cups/$(LIBCUPS)
 
 parallel:	parallel.o ../cups/$(LIBCUPS) libbackend.a
 	echo Linking $@...
-	$(CC) $(LDFLAGS) -o parallel parallel.o libbackend.a $(LIBS)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o parallel parallel.o libbackend.a $(LIBS)
 
 
 #
@@ -169,7 +169,7 @@ parallel:	parallel.o ../cups/$(LIBCUPS) 
 
 scsi:	scsi.o ../cups/$(LIBCUPS)
 	echo Linking $@...
-	$(CC) $(LDFLAGS) -o scsi scsi.o $(LIBS)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o scsi scsi.o $(LIBS)
 
 scsi.o:	scsi.c scsi-irix.c scsi-linux.c
 
@@ -180,7 +180,7 @@ scsi.o:	scsi.c scsi-irix.c scsi-linux.c
 
 serial:	serial.o ../cups/$(LIBCUPS) libbackend.a
 	echo Linking $@...
-	$(CC) $(LDFLAGS) -o serial serial.o libbackend.a $(BACKLIBS) $(LIBS)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o serial serial.o libbackend.a $(BACKLIBS) $(LIBS)
 
 
 #
@@ -189,7 +189,7 @@ serial:	serial.o ../cups/$(LIBCUPS) libb
 
 snmp:	snmp.o ../cups/$(LIBCUPS) libbackend.a
 	echo Linking $@...
-	$(CC) $(LDFLAGS) -o snmp snmp.o libbackend.a $(LIBS)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o snmp snmp.o libbackend.a $(LIBS)
 
 
 #
@@ -198,7 +198,7 @@ snmp:	snmp.o ../cups/$(LIBCUPS) libbacke
 
 socket:	socket.o ../cups/$(LIBCUPS) libbackend.a
 	echo Linking $@...
-	$(CC) $(LDFLAGS) -o socket socket.o libbackend.a $(LIBS)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o socket socket.o libbackend.a $(LIBS)
 
 
 #
@@ -207,7 +207,7 @@ socket:	socket.o ../cups/$(LIBCUPS) libb
 
 usb:	usb.o ../cups/$(LIBCUPS) libbackend.a
 	echo Linking $@...
-	$(CC) $(LDFLAGS) -o usb usb.o libbackend.a $(BACKLIBS) $(LIBS)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o usb usb.o libbackend.a $(BACKLIBS) $(LIBS)
 usb.o:	usb.c usb-darwin.c usb-unix.c
 
 
